xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp

```diff
@@ -1068,10 +1069,9 @@ CDVDVideoCodec::VCReturn CDVDVideoCodecAndroidMediaCodec::GetPicture(VideoPictur
     else
     {
       m_videobuffer.videoBuffer = nullptr;
-      if (retgp == -1 || ((m_codecControlFlags & DVD_CODEC_CTRL_DRAIN)!=0 && ++m_noPictureLoop == 10))  // EOS
+      if (retgp == -1 || m_state == MEDIACODEC_STATE_WAIT_ENDOFSTREAM) // EOS
```

Comment:
now CJNIMediaCodec::INFO_TRY_AGAIN_LATER will result in eof too.

Reply:
This is intended, timeout is set to 100ms.

Reply:
right, might work. somehow :) nevertheless, it does not properly implement the sates in the docs. In case of error the final state should be ERROR not EOS. bringing the codec from ERROR back to FLUSHED requires a different approach than from EOS to FLUSHED.


![image](https://user-images.githubusercontent.com/576009/84167098-172e6200-aa76-11ea-9e09-a5743f30586c.png)

Reply:
Yes, you are totally right. ERROR is set now. In the next iteration I'll have to look for a proper error handling, MediaCodec has recoverable and non-recoverable errors which are not yet interpreted from JNI (I'm even not sure if the JNI interface alllows such detailed level)

Thx much for looking over the code1